mautrix_telegram/__main__.py

```diff
@@ -137,6 +138,19 @@ async def start(self) -> None:
                 self._loop_check_as_connection_pool()
             )
 
+        if not self.config["telemetry.enabled"]:
+            self.log.warning(
+                "** "
+                "Telemetry is disabled in config. "
+                "This may violate your terms of service if not expressly permitted under license with New Vector Ltd. "
+                "Please contact ems-support@element.io with any questions, or details on how to silence this warning. "
+                "**"
+            )
+        else:
+            # Get an identifier for the current instance (used for licensing / telemetry)
+            self.config["telemetry.instance_id"] = instance_id = get_instance_id(self.log)
```

Comment:
This flow was based on the Teams bridge, which looks up the ID from a file, which is generated if it doesn't exist.  Should the generated ID be written back to `config.yaml`, or is it enough for it to be in a standalone file?  (This question applies to the Teams bridge too, I suppose...)